### PR TITLE
Restore Membership for User in Grace Period

### PR DESCRIPTION
**Pull request by RobertCram**
#### PT URL: https://www.pivotaltracker.com/story/show/178579872

In light of the current payment problems it has become important for the admin to be able to restore the membership for a user in the grace period.

## Changes proposed in this pull request:
1. Enable the status change in_grace_period => current_member, with all that that entails.
2. Increase the Capybara.default_max_wait_time to 15 seconds, so feature tests can also pass on slower machines.

## Additional Info:
When extending the end date for a user in a grace period, theoretically there are two possibilities: 
1. Extending to a date in the past. This will only change the lastdate of the membership, but will not change the users status. 
2. Extending to a date in the future. This will change the lastdate of the membership, and will also change the users status to "current_member", triggering any (email) actions that are defined in the RestoreIndividualmemberActions class (currently none)

_Note: I'm using the existing code for **user.current_member?**, but that seems to have a one-off error. This means that a membership with the current date as lastdate will be seen as expired. Changing that is out of the scope of this PR._

---
## Ready for review:
@patmbolger @thesuss

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -108,6 +108,12 @@ def edit_status
         if last_day_param != current_membership.last_day
           admin_change_note << change_membership_last_day_and_note(current_membership, last_day_param)
         end
+      elsif @user.in_grace_period?
+        current_membership = @user.most_recent_membership
+        admin_change_note << t('memberships.auto_added_notes.started_on', first_day: Date.current )
```
Comment (patmbolger) [inline app/controllers/users_controller.rb]: Here, we're not actually changed the membership _start_ date, so this message is misleading.

**Review by patmbolger — CHANGES_REQUESTED** [review]
Looks great!  Please see comment.

**Comment by patmbolger** [comment]
This is not an action required for this PR, but when reviewing this I noticed that we are not updating the member status info in the view, once the controller has completed the action.

Here is the view that I see _after_ extending the grace-period member's membership:

Screen Shot 2022-01-17 at 7 22 54 AM

Whereas, if I refresh the view, this is what I should have seen:

Screen Shot 2022-01-17 at 7 34 36 AM

The JS callback 
```
$("body").on("ajax:success", "#user-status-form", function(e, data) { ....
```
is supposed to update the view with the HTML returned by the controller action.  Looking at the history os changes in the view and JS function, that auto refresh of the view seems to have been broken for quite a while.

Anyway, this does not affect this PR, _and_ a fix for this would be low priority anyway since it just affects what the _admin_ sees.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (RobertCram) [reply app/controllers/users_controller.rb]: Good catch. Copy paste error. I'll fix it.

**Comment by RobertCram** [comment]
I noticed the view not updating and also decided it was not in the scope of this PR. Planned on fixing it in another one. Thanks for the hint on where to look for the problem.

**Pull request merged**